### PR TITLE
Update BCL package versions

### PR DESCRIPTION
**Pull request by m-redding**
Upgrades versions for a handful of BCL packages to either latest or 6.x (whichever is lower). This is in preparation for updating System.Text.Json, which has dependencies on System.Text.Encodings.Web and Microsoft.Bcl.AsyncInterfaces.

### CODE REVIEW

File: sdk/core/Azure.Core/src/Azure.Core.csproj [hunk 1]
```diff
@@ -19,15 +19,18 @@
     <SupportedPlatform Include="browser" />
   </ItemGroup>
 
-  <ItemGroup>
-    <PackageReference Include="Microsoft.Bcl.AsyncInterfaces" />
-    <PackageReference Include="System.ClientModel" />
+  <ItemGroup Condition="'$(TargetFramework)' != 'net6.0' ">
```
Comment (jsquire) [inline sdk/core/Azure.Core/src/Azure.Core.csproj:22]: Is this an interim step before moving the "strip polyfill packages" step in the build targets?

Comment (m-redding) [reply sdk/core/Azure.Core/src/Azure.Core.csproj:22]: I'll take this out - was thinking this might be an easy win but can come back to the build targets

**Review by jsquire — APPROVED** [review]

**Comment by m-redding** [comment]
The storage test pipeline is failing with:
```
```

**Pull request merged**